src/dag.rs

```diff
@@ -125,4 +125,240 @@ where
 }
 
 #[cfg(test)]
-mod tests {}
+mod tests {
+    use super::*;
+    use petgraph::graph::DiGraph;
+    use std::collections::HashSet;
+
+    #[test]
+    fn test_build_dag_success() {
+        // Test: Can we build a dag?
+        let nodes: HashSet<i32> = [1, 2, 3, 4].iter().cloned().collect();
```

Comment:
@coderabbitai Please use `.copied()` instead of `.cloned()` when we can. Apply throughout the dag unit tests.